### ISSUE TITLE
Crashes in code viewer

### ISSUE THREAD

**Issue by HGustavs** [HGustavs/LenaSYS#2304]
In some views the code viewer crashes when you click the cog wheel

The culprit seems to be these lines: 

```
box[5]=box[5].replace(/&#47;/g, "/"); 
$("#filename").val(box[5]);
```

In some cases there is no box[5] in the data returned from the service.

A check to make sure that box[5] exists and only execute the code if it does, and to perform some other alternative code if it does not, will hopefully resolve this issue?

**Comment by a14jonal** [comment]
I can't reproduce the error can you post the steps you made in order to get this crash. Some things to note is that before the most recent merge there was no template that had 5 boxes, and the code you showed is older than that change.

**Comment by HGustavs** [comment]
Perhaps the code does not exist in more recent branches?

If the code does exist it is just necessary to check if box[5] exists, if the code has been removed we can just retire this issue.

**Comment by a14jonal** [comment]
The code still exist but I am unsure how you get the code viewer to crash. Box[5] is also referenced in the updateContent() function.
~~~
if (box[1] != $("#boxcontent").val() || box[3] != $("#wordlist").val() || box[4] != $("#boxtitle").val() || box[5] != $("#filename option:selected").val() || addedRows.length > 0 || removedRows.length > 0) {
~~~

**Comment by HGustavs** [comment]
I was thinking something along the lines of:

~~~
if(box[5]!=null)[
    box[5]=box[5].replace(/&#47;/g, "/"); 
    $("#filename").val(box[5]);
}else{
    $("#filename").val("No file");
}
~~~

**Comment by a17joaap** [comment]
This was fixed in 9d27b9fb

**Comment by a17alevu** [comment]
Closing.
